### PR TITLE
Improvement: Add configuration "Infill Pattern" and "Fan Speed" to 3DP settings

### PR DESCRIPTION
**Pull request by bbaa77770**
LU-716: Add "Infill Pattern" in printing settings
LU-579: Add "Fan Speed" in material settings

### CODE REVIEW

File: src/app/views/PrintingManager/PrintingManager.jsx [hunk 1]
```diff
@@ -33,7 +33,8 @@ const QUALITY_CHECKBOX_AND_SELECT_KEY_ARRAY = [
     'retraction_hop_enabled',
     'adhesion_type',
     'support_type',
-    'support_pattern'
+    'support_pattern',
+    'infill_pattern'
```
Comment (parachvte) [inline src/app/views/PrintingManager/PrintingManager.jsx:37]: TODO

File: src/app/constants/index.js [hunk 2]
```diff
@@ -106,8 +108,8 @@ export const PRINTING_QUALITY_CONFIG_KEYS = [
     'support_angle',
     'support_pattern',
     'support_infill_rate',
-    'support_z_distance'
-
+    'support_z_distance',
+    'cool_fan_speed'
```
Comment (parachvte) [inline src/app/constants/index.js:112]: TODO

File: resources/CuraEngine/Config/snapmaker.def.json [hunk 3]
```diff
@@ -50,6 +58,10 @@
             "label": "Infill Density",
             "description": "Adjusts the density of infill of the print."
         },
+        "infill_pattern": {
+            "label": "Infill Pattern",
+            "description": "The pattern of the infill material of the print. The line and zig zag infill swap direction on alternate layers, reducing material cost. The grid, triangle, tri-hexagon, cubic, octet, quarter cubic, cross and concentric patterns are fully printed every layer. Gyroid, cubic, quarter cubic and octet infill change with every layer to provide a more equal distribution of strength over each direction."
```
Comment (parachvte) [inline resources/CuraEngine/Config/snapmaker.def.json:63]: Add options here for i18n extracting.

**Review by parachvte — APPROVED** [review]

**Review by meiliang1123 — APPROVED** [review]
LGTM

**Pull request merged**